spec/models/post_spec.rb

```diff
@@ -1,5 +1,43 @@
 require 'rails_helper'
 
 RSpec.describe Post, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  user = User.create(name: 'Tom', posts_counter: 0)
+  subject do
+    Post.new(author_id: user.id, title: 'Hello', text: 'This is my first post', likes_counter: 0, comments_counter: 0)
+  end
+
+  before(:each) { subject.save }
+
+  it 'check if the title is not blank' do
+    subject.title = ''
+    expect(subject).to_not be_valid
+  end
+
+  it 'likes_counter must be not <= 0 failure' do
+    subject.likes_counter = nil
+    expect(subject).to_not be_valid
+  end
+  it 'likes_counter must be not <= 0 success' do
+    expect(subject).to be_valid
+  end
+
+  it 'comments_counter must be not <= 0 failure' do
+    subject.comments_counter = nil
+    expect(subject).to_not be_valid
+  end
+  it 'comments_counter must be not <= 0 success' do
+    expect(subject).to be_valid
+  end
+
+  it 'return the recent comments for a post' do
+    expect(Post.five_recent_comments(subject).count).to eql 0
+  end
+
+  it 'return the recent comments for a post' do
+    expect(Post.update_user_posts_counter(user)).to eql true
+  end
+
+  it 'return all the comments for a post' do
+    expect(Post.all_posts_for_a_user(subject.id).count).to eql 0
+  end
```

Comment:
- [ ] <b>REQUIRED:</b> Only this test case is failing, Kindly address the issue and fix it. 
![Screen Shot 2022-03-30 at 22 45 43](https://user-images.githubusercontent.com/45217364/160928469-c3a58425-11ca-4d6c-ac16-bcbe56e22efe.png)